src/compiler/checker.ts

```diff
@@ -5625,6 +5624,10 @@ namespace ts {
             return checkTypeComparableTo(source, target, /*errorNode*/ undefined);
         }
 
+        function areTypesComparable(type1: Type, type2: Type): boolean {

```

Comment:
:100: 

Add a comment that this relationship is bidirectional and doesn't report errors.